### PR TITLE
fix(apigatewayv2): authorizer is not removed when HttpNoneAuthorizer is used

### PR DESCRIPTION
**Pull request by njlaw**
CloudFormation will not remove an existing Authorizer if AuthorizationType and AuthorizerId are simply removed.  The AuthorizationType must be explicitly set to NONE for CloudFormation to remove the existing Authorizer.

As such, I updated the HttpRoute constructor to include the AuthorizationType even if it is NONE; otherwise it is impossible to remove an authorizer in CDK.  Some thought had obviously gone into this previously because of the following line:

https://github.com/aws/aws-cdk/blob/2f5eeb08f8790c73db7305cc7f85116e2730267d/packages/%40aws-cdk/aws-apigatewayv2/lib/http/route.ts#L159

I did not manage to track down the reasoning for this in commit comments, so I would be interested to hear why this was done, since I may have overlooked a desired use case.  I'm wondering if it was assumed that the default CloudFormation value for AuthorizationType is NONE, so to have a more compact template it was omitted.  However, the behavior when AuthorizationType is not present, is to not change the existing Authorizer.

Basically in the CloudFormation template,

```yaml
  APIGETintegrationgoogleapiregister1D8736BD:
    Type: AWS::ApiGatewayV2::Route
    Properties:
      ApiId:
        Ref: API62EA1CEE
      RouteKey: GET /integration/google-api/register
      Target: ...
```

does not have the same effect as

```yaml
  APIGETintegrationgoogleapiregister1D8736BD:
    Type: AWS::ApiGatewayV2::Route
    Properties:
      ApiId:
        Ref: API62EA1CEE
      RouteKey: GET /integration/google-api/register
      AuthorizationType: NONE
      Target: ...
```

Only the later will remove an existing Authorizer.

If you think this is a bug in CloudFormation and not its intended behavior, please let me know.  I am assuming that they would not change the behavior anyway since that could have unintended consequence for anyone who redeploys a template without the AuthorizationType set.

BREAKING CHANGE: setting the authorizer of an API route to HttpNoneAuthorizer will now remove any existing authorizer on the route

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-apigatewayv2/lib/http/route.ts [hunk 1]
```diff
@@ -167,7 +165,7 @@ export class HttpRoute extends Resource implements IHttpRoute {
       routeKey: props.routeKey.key,
       target: `integrations/${integration.integrationId}`,
       authorizerId: authBindResult?.authorizerId,
-      authorizationType,
+      authorizationType: authBindResult?.authorizationType,
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigatewayv2/lib/http/route.ts:168]: To capture all cases and to safeguard against a future change where `authorizationType` from the bind result is made an optional prop, we should default to 'NONE' here.

```suggestion
      authorizationType: authBindResult?.authorizationType ?? HttpAuthorizerType.NONE, // must be explicitly NONE (not undefined) for stack updates to work correctly
```

**Review by nija-at — CHANGES_REQUESTED** [review]
Thanks for submitting this PR. The approach sounds good.

One suggestion below

File: packages/@aws-cdk/aws-apigatewayv2/lib/http/route.ts (hunk 1 above)
Comment (njlaw) [reply packages/@aws-cdk/aws-apigatewayv2/lib/http/route.ts:168]: I'm wondering if there is any possible instance where someone would specifically want the `authorizationType` to be undefined as that is an option on the CloudFormation side.  If that actually is desired, the user could always `.addPropertyOverride('AuthorizationType', undefined)` to the resource node.

File: packages/@aws-cdk/aws-apigatewayv2/lib/http/route.ts [hunk 2]
```diff
@@ -167,7 +165,7 @@ export class HttpRoute extends Resource implements IHttpRoute {
       routeKey: props.routeKey.key,
       target: `integrations/${integration.integrationId}`,
       authorizerId: authBindResult?.authorizerId,
-      authorizationType,
+      authorizationType: authBindResult?.authorizationType ?? HttpAuthorizerType.NONE, // must be explicitly NONE (not undefined) for stack updates to work correctly
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigatewayv2/lib/http/route.ts:168]: Could you also update this test to assert that the default is set to NONE - https://github.com/aws/aws-cdk/blob/f0d5c25e1ae026eef03dc396e48368521dcb8331/packages/%40aws-cdk/aws-apigatewayv2/test/http/route.test.ts#L19

**Review by nija-at — CHANGES_REQUESTED** [review]
The build has failed. The link to the build logs are in a comment above.

**Review by nija-at — CHANGES_REQUESTED** [review]
Hi @njlaw - There build is still failing. Likely more snapshot tests that need to be updated.

**Review by nija-at — APPROVED** [review]
